src/Parser.js

```diff
@@ -653,18 +672,24 @@ export default class Parser {
 
     /**
      * Call a function handler with a suitable context and arguments.
-     * @param {string} name
-     * @param {Array<ParseNode>} args
-     * @param {Array<?ParseNode>} optArgs
-     * @param {Token=} token
      */
-    callFunction(name, args, optArgs, token) {
-        const context = {
+    callFunction(
+        name: string,
+        args: ParseNode[],
+        optArgs: (?ParseNode)[],
+        token?: Token,
+    ) {
```

Comment:
Return type is missing. I'm unsure, but it might have to be something generic like `*` before the `ParseNode` refactor.